frame/democracy/src/lib.rs

```diff
@@ -659,6 +677,25 @@ pub mod pallet {
 			Ok(())
 		}
 
+        #[pallet::weight(T::WeightInfo::promote())]
+        pub fn promote(
+            origin: OriginFor<T>,
+            proposal_hash: T::Hash,
+        ) -> DispatchResult {
+            let who = T::PromotionOrigin::ensure_origin(origin)?;
+
+            let public_props = PublicProps::<T>::get();
+            public_props.iter().position(|p| p.1 == proposal_hash).ok_or_else(|| Error::<T>::ProposalMissing)?;
+
+            let mut promoted_by = <Promoted<T>>::get(&proposal_hash).unwrap_or_else(Vec::new);
+            let insert_positions = promoted_by.binary_search(&who).err().ok_or(Error::<T>::AlreadyPromoted)?;
+            promoted_by.insert(insert_positions, who.clone());
+            <Promoted<T>>::insert(&proposal_hash, promoted_by);
+            Self::deposit_event(Event::<T>::PromotedBy { proposal_hash, who });
+
+            Ok(())
```

Comment:
What happens after something is promoted?

How does this storage item you are creating get cleaned up?

Reply:
It is cleaned up when a proposal gets moved to the external queue and should also be cleaned when canceled and blacklisted